sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_responses.py

```diff
@@ -0,0 +1,38 @@
+# The MIT License (MIT)
+# Copyright (c) 2024 Microsoft Corporation
+
+from azure.core.utils import CaseInsensitiveDict
+
+
+class CosmosDictResponse(dict):
+    def __init__(self, original_dict, response_headers):
+        if original_dict is None:
+            original_dict = {}
+        super().__init__(original_dict)
+        self._response_headers = response_headers
```

Comment:
You may need to actually wrap this in CaseInsensitiveDict is it's not one already, and if it always is one already, it should be properly documented as such in the class docstring and constructor type hint.